src/managedwriter/writer_client.ts

```diff
@@ -0,0 +1,418 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as gax from 'google-gax';
+import type {CallOptions, ClientOptions} from 'google-gax';
+import * as protos from '../../protos/protos';
+
+import {BigQueryWriteClient} from '../v1';
+
+/**
+ *  BigQuery Write API.
+ *
+ *  The Write API can be used to write data to BigQuery.
+ *
+ *  For supplementary information about the Write API, see:
+ *  https://cloud.google.com/bigquery/docs/write-api
+ * @class
+ * @memberof storage
+ */
+
+type StreamConnection = {
+  write_stream: WriteStream;
+  connection: gax.CancellableStream;
+};
+type streamConnectionsMap = Record<string, ManagedStream>;
+type StreamConnections = {
```

Comment:
unclear from this how connections are associated with managedstreams/writers.  the map is keyed by stream id?

Reply:
yeah, they are keyed by `streamId`. I'll more docs to those types.